### DISCUSSION TITLE
Don't understand data distribution

### DISCUSSION THREAD

**Discussion by Quiark** [minio/minio#13648] [Q&A] [answered]
Hi,

When specifying drives using `minio serve`, is there any special meaning to the order of drives or the 1...4 syntax (other than auto-expanding)? If I have multiple drives on one machine, how do I know among which machines is a file going to be split and what if MinIO ends up selecting a data erasure set that is actually only on one machine (and the machine then goes up in flames?).

Thanks

**Comment by klauspost** [comment, marked as answer]
Yes, drives should retain order. Each disk has metadata that describes its place in the cluster.

> how do I know among which machines is a file going to be split and what if MinIO ends up selecting a data erasure set that is actually only on one machine (and the machine then goes up in flames?).

Sets (with objects) are always distributed across servers first, so each server will hold as few shards of each object as possible in your setup, thereby limiting impact of a downed server.
